src/libxrpl/protocol/TxFormats.cpp

```diff
@@ -47,472 +47,18 @@ TxFormats::TxFormats()
-            {sfAmount, soeREQUIRED},
-        },
-        commonFields);
-
-    add(jss::XChainAddClaimAttestation,
-        ttXCHAIN_ADD_CLAIM_ATTESTATION,
-        {
-            {sfXChainBridge, soeREQUIRED},
-
-            {sfAttestationSignerAccount, soeREQUIRED},
-            {sfPublicKey, soeREQUIRED},
-            {sfSignature, soeREQUIRED},
-            {sfOtherChainSource, soeREQUIRED},
-            {sfAmount, soeREQUIRED},
-            {sfAttestationRewardAccount, soeREQUIRED},
-            {sfWasLockingChainSend, soeREQUIRED},
-
-            {sfXChainClaimID, soeREQUIRED},
-            {sfDestination, soeOPTIONAL},
-        },
-        commonFields);
-
-    add(jss::XChainAddAccountCreateAttestation,
-        ttXCHAIN_ADD_ACCOUNT_CREATE_ATTESTATION,
-        {
-            {sfXChainBridge, soeREQUIRED},
-
-            {sfAttestationSignerAccount, soeREQUIRED},
-            {sfPublicKey, soeREQUIRED},
-            {sfSignature, soeREQUIRED},
-            {sfOtherChainSource, soeREQUIRED},
-            {sfAmount, soeREQUIRED},
-            {sfAttestationRewardAccount, soeREQUIRED},
-            {sfWasLockingChainSend, soeREQUIRED},
-
-            {sfXChainAccountCreateCount, soeREQUIRED},
-            {sfDestination, soeREQUIRED},
-            {sfSignatureReward, soeREQUIRED},
-        },
-        commonFields);
-
-    add(jss::XChainAccountCreateCommit,
-        ttXCHAIN_ACCOUNT_CREATE_COMMIT,
-        {
-            {sfXChainBridge, soeREQUIRED},
-            {sfDestination, soeREQUIRED},
-            {sfAmount, soeREQUIRED},
-            {sfSignatureReward, soeREQUIRED},
-        },
-        commonFields);
-
-    add(jss::DIDSet,
-        ttDID_SET,
-        {
-            {sfDIDDocument, soeOPTIONAL},
-            {sfURI, soeOPTIONAL},
-            {sfData, soeOPTIONAL},
-        },
-        commonFields);
-
-    add(jss::DIDDelete, ttDID_DELETE, {}, commonFields);
-
-    add(jss::OracleSet,
-        ttORACLE_SET,
-        {
-            {sfOracleDocumentID, soeREQUIRED},
-            {sfProvider, soeOPTIONAL},
-            {sfURI, soeOPTIONAL},
-            {sfAssetClass, soeOPTIONAL},
-            {sfLastUpdateTime, soeREQUIRED},
-            {sfPriceDataSeries, soeREQUIRED},
-        },
-        commonFields);
-
-    add(jss::OracleDelete,
-        ttORACLE_DELETE,
-        {
-            {sfOracleDocumentID, soeREQUIRED},
-        },
-        commonFields);
-
-    add(jss::LedgerStateFix,
-        ttLEDGER_STATE_FIX,
-        {
-            {sfLedgerFixType, soeREQUIRED},
-            {sfOwner, soeOPTIONAL},
-        },
-        commonFields);
+#pragma push_macro("UNWRAP")
+#undef UNWRAP
+#define UNWRAP(...) __VA_ARGS__
+#pragma push_macro("TRANSACTION")
+#undef TRANSACTION
+#define TRANSACTION(tag, value, name, fields) \
+    add(jss::name, tag, UNWRAP fields, commonFields);
+#include <xrpl/protocol/transactions.h>
+#undef TRANSACTION
+#pragma pop_macro("TRANSACTION")
+#undef UNWRAP
+#pragma pop_macro("UNWRAP")
```

Comment:
For consistency, please follow the same style as in Feature.cpp (vertical spacing)